### PR TITLE
Builds marked as aborted, rather than failed

### PR DESCRIPTION
**Pull request by sturivny**

### CODE REVIEW

**Comment by msrb** [comment]
Since you're already importing the pipeline library, you can use the `abort()` step: https://github.com/fedora-ci/jenkins-pipeline-library/blob/master/docs/steps/abort.md

**Comment by bgoncalv** [comment]
When we sent status as aborted, will it send ci message as `failed`?

**Comment by bgoncalv** [comment]
Note, before it was just skipping the next steps so the build was marked as passed.

File: Jenkinsfile [hunk 1]
```diff
@@ -253,7 +255,8 @@ pipeline {
             steps {
                 script {
                     if (!hasTests || env.namespace == "tests") {
-                        return
+                        currentBuild.result = 'ABORTED'
+                        error('nvr verify failed. Aborted.')
```
Comment (bgoncalv) [inline Jenkinsfile:259]: this is not a failure.
For example, when running on tests namsepace, no extra rpm is installed to the qcow2, so we just skip nvr verify, but this is not an error.

**Comment by sturivny** [comment]
@msrb @bgoncalv Updated PR. 
Please, take a look :)

File: Jenkinsfile [hunk 2]
```diff
@@ -179,6 +179,10 @@ pipeline {
                     test_playbooks = result["test_playbooks"]
                     if (test_playbooks.size() > 0) {
                         hasTests = true
+                    } else if (test_playbooks.size() == 0) {
```
Comment (bgoncalv) [inline Jenkinsfile:182]: much better, but I think the test size > 0 is not needed.
You can just abort if size == 0, no other test should be needed here, like negative size will never happen :)
Also, as we will abort if there is no tests, the hasTests variable becomes unnecessary.

File: Jenkinsfile [hunk 3]
```diff
@@ -322,6 +326,13 @@ pipeline {
                 }
             }
         }
+        aborted {
+            script {
+                if (artifactId) {
+                    sendMessage(type: 'complete', artifactId: artifactId, pipelineMetadata: pipelineMetadata, xunit: xunit, dryRun: params.dryRun)
```
Comment (bgoncalv) [inline Jenkinsfile:332]: I'd send message with type `error` instead of `complete`.

Comment (sturivny) [reply Jenkinsfile]: done

**Pull request merged**